src/DeploymentProvider/DeploymentHelper.ts

```diff
@@ -306,6 +253,76 @@ export class DeploymentHelper {
+        }
+        if (params.buildMemory) {
+            build.properties.resourceRequests.memory = params.buildMemory;
+        }
+        let transformedBuildEnvironmentVariables = {};
+        if (params.buildEnv) {
+            core.debug("Build environment variables modified.");
+            const parsedBuildEnvVariables = parse(params.buildEnv);
+            //Parsed pairs come back as  {"key1":{"value":"val1"},"key2":{"value":"val2"}}
+            Object.keys(parsedBuildEnvVariables).forEach(key => {
+                transformedBuildEnvironmentVariables[key] = parsedBuildEnvVariables[key]['value'];
+            });
+            build.properties.env = transformedBuildEnvironmentVariables;
+        }
+        core.debug('build: ' +  JSON.stringify(build));
+        const buildResponse = await client.buildServiceOperations.createOrUpdateBuild(params.resourceGroupName, params.serviceName, this.buildServiceName, buildName, build);
+        core.debug('build response: ' +  JSON.stringify(buildResponse));
+        const regex = RegExp("[^/]+$");
+        const buildResultName = regex.exec(buildResponse.properties.triggeredBuildResult.id)[0];
+        let buildProvisioningState = 'Queuing';
+        let cnt = 0;
+        core.debug("wait for build result......");
+        //Waiting for build result. Timeout 30 minutes.
+        const logStream = await this.logStreamConstructor(client, params);
+        const stagesRecorded = new Set();
+        while (buildProvisioningState != 'Succeeded' && buildProvisioningState != 'Failed' && cnt++ < 180) {
+            await new Promise(resolve => setTimeout(resolve, 10000));
+            core.debug("wait for 10 seconds....");
+            try {
+                const waitResponse = await client.buildServiceOperations.getBuildResult(params.resourceGroupName, params.serviceName, this.buildServiceName, buildName, buildResultName);
+                waitResponse.properties.buildStages.forEach(async stage => {
+                    if (!stagesRecorded.has(stage.name) && stage.status != "NotStarted") {
+                        const url=`https://${logStream["baseUrl"]}/api/logstream/buildpods/${waitResponse.properties.buildPodName}/stages/${stage.name}?follow=true`
+                        const credentials = Buffer.from(`primary:${logStream["primaryKey"]}`).toString('base64');
+                        const auth = { "Authorization" : `Basic ${credentials}` };
+                        const response = await fetch(url, {method: 'GET', headers : auth });
+                        response.body.pipe(process.stdout);
+                        stagesRecorded.add(stage.name);
+                    }
+                });
+                core.debug('build result response: ' + JSON.stringify(waitResponse));
+                buildProvisioningState = waitResponse.properties.provisioningState;
+            }
+            catch (e:any) {
+                console.log(e.message);
```

Comment:
Should we use `core.xxx` here for logging?

Reply:
For this part, I think console.log() equals core.info(). In the [tutorial](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#writing-the-action-code), it also uses console.log() to print logs. If we use core.error(), it may cause some misunderstanding when the exception caught. It will produce [annotations](https://github.com/actions/toolkit/tree/main/packages/core#annotations) but this is only some errors when waiting the build response. I assume it maybe some network issues. So I use console.log() or maybe we can use core.info().